Added ⚡️ content to to `Onboarding` >  `Funding a wallet` (#569)
BitcoinDesign/Guide: guide/onboarding/funding-a-wallet.md

# Funding a wallet

The final stage of onboarding users to a Bitcoin wallet is having them fund it with some bitcoin. Below we explore different design directions your application could take when users fund their wallet for the first time.

## With Lightning services

A "Lightning service" is a service wallet providers offer users to avoid having them manage and configure their wallets manually. Some of these services, and the user pain points they solve when funding a wallet, are as followed:

* Pay-to-open provides on-demand inbound payment channel liquidity so users can fund their wallet with Lightning payments straight away.
* Swaps so users can open a payment channel and fund their wallet with an on-chain transfer. 

> * Spend-unconfirmed so users can send and receive as soon as their wallet is funded without having to wait for the on-chain transaction to confirm.

Reviewer: I think zero confirmation is clearer here? And it's already used term in wallets like Breez.
```suggestion
* Zero-confirmation so users can send and receive as soon as their wallet is funded without having to wait for the on-chain transaction to confirm.
```

Author: They call them turbo channels in Breeze, which I'm not a fan. Zero-confirmation doesn't really give much of a description to what that service is offering. Spend-unconfirmed is used by Phoenix and I think is a bit more descriptive to what the service is actually enabling.

   retina = "/assets/images/guide/onboarding/funding-a-wallet/lightning-services-invoice@2x.png"
   layout = "shadow"
   caption = "The first time the user funds their wallet a payment channel will be opened."
   alt-text = "Screen showing funding the wallet with an invoice."
   width = 250
   height = 541
%}

{% include picture.html
   image = "/assets/images/guide/onboarding/funding-a-wallet/lightning-services-address.png"
   retina = "/assets/images/guide/onboarding/funding-a-wallet/lightning-services-address@2x.png"
   class = "shadow"
   caption = "The user should have the option to fund via an address, as they may not have funds in Lightning."
   alt-text = "Screen showing funding the wallet with an address."
   width = 250
   height = 541
%}

{% include picture.html
     image = "/assets/images/guide/onboarding/funding-a-wallet/lightning-services-final.png"
   retina = "/assets/images/guide/onboarding/funding-a-wallet/lightning-services-final@2x.png"
   class = "shadow"
   caption = "Once the wallet has been funded it is ready to use."
   alt-text = "Screen showing the wallet funded and ready to use."
   width = 250
   height = 541
%}

</div>

> The initial screen when the user opens their wallet for the first time should prompt the user to fund their wallet so they can begin using it. Depending on how your application connects to the Bitcoin and Lightning network you may need to notify the user that funds will not appear until these connections are established and syncd. We recommend connecting to the Bitcoin network via Neutrino which offers a simple, private way to source blockchain data.

Reviewer 2: Another thought. Should this be "a client like Neutrino" or "block filters"? I raise this point because I like to adhere to a principle of avoiding engineering recommendations in the Guide. From a design perspective, we recommend principles like privacy, transparency, interoperability, etc. I feel that how the bitcoin and lightning engineers accomplish these principles is for them to decide. Thoughts, anyone?

> The initial screen when the user opens their wallet for the first time should prompt the user to fund their wallet so they can begin using it. Depending on how your application connects to the Bitcoin and Lightning network you may need to notify the user that funds will not appear until these connections are established and syncd. We recommend connecting to the Bitcoin network via Neutrino which offers a simple, private way to source blockchain data.

Author: I agree, how about saying something like below: 

Common ways applications connect to the Bitcoin network is using compact block filters (BIP 157/158) or SPV to query and download block headers. Block filters offer more privacy but take some initial time to sync whilst an SPV does not offer any privacy to users but is ready instantly. 

Can link to the entries for these in the `Nodes` page.

   retina = "/assets/images/guide/onboarding/funding-a-wallet/without-services-channel@2x.png"
   class = "shadow"
   caption = "Once the wallet on-chain funds are confirmed, a payment channel is opened."
   alt-text = "Screen showing a channel being ready to be opened."
   width = 250
   height = 541
%}

{% include picture.html
     image = "/assets/images/guide/onboarding/funding-a-wallet/without-services-final.png"
   retina = "/assets/images/guide/onboarding/funding-a-wallet/without-services-final@2x.png"
   class = "shadow"
   caption = "Once a channel has been opened the wallet is ready to use."
   alt-text = "Screen showing the wallet funded and ready to use."
   width = 250
   height = 541
%}

</div>

The initial screen when the user opens their wallet for the first time should prompt the user to fund their wallet so they can begin using it. Depending on how your application connects to the Bitcoin and Lightning network you may need to notify the user that funds will not appear until these connections are established and syncd. We recommend connecting to the Bitcoin network via Neutrino which offers a simple, private way to source blockchain data.

> When the user funds their wallet without Lightning services the user needs to send sats on-chain first then open a payment channel with those sats. The user should be prompted that they should send enough sats to cover the transaction fee for opening a payment channel. 

Reviewer 2: ```suggestion
When the user funds their wallet without Lightning services, they need to send sats on-chain first and then use the sats to open a payment channel. They should be prompted to send enough sats to cover the transaction fee for opening a payment channel. 
```

> For a simplified UX, the channel open could be initiated as soon as on-chain funds are confirmed so users don’t have to manually open a channel. Your wallet application could have a node that automatically opens a channel with the user. Having your application as a dedicated peer means users are less likely to have channels closed on them for inactivity. A wallet that has automatic channel opens to their own dedicated node is [Blixt](https://blixtwallet.github.io/).

Reviewer 2: ```suggestion
For a simplified UX, the channel open could be initiated as soon as on-chain funds are confirmed so users don’t have to manually open a channel. Your wallet application could have a node that automatically opens a channel with the user. Having your application as a dedicated peer means users are less likely to have channels closed on them for inactivity. [Blixt](https://blixtwallet.github.io/) is an example of a wallet that has automatic channel opens to their own dedicated node.
```

Consistency. When you mentioned Phoenix, it was "an example".

   alt-text = "Screen showing user manually opening a channel."
   width = 250
   height = 541
%}

{% include picture.html
     image = "/assets/images/guide/onboarding/funding-a-wallet/manual-channel-final.png"
   retina = "/assets/images/guide/onboarding/funding-a-wallet/manual-channel-final@2x.png"
   class = "shadow"
   caption = "Once a channel has been opened the wallet is ready to use."
   alt-text = "Screen showing the wallet funded and ready to use."
   width = 250
   height = 541
%}

</div>

Alternatively users could manually source their own peers to open channels with. You may want to offer a directory to assist users with finding peers.

Opening channels without Lightning services like pay-to-open means the user will have no inbound liquidity to receive payments. Users will only be able to send initially and will only be able to receive once they have sent some payments. This should be clearly communicated to the user. We will explore this in more detail in the payments chapter. 

You may want to offer services in your directory like LNBig that opens channels with inbound liquidity for users so they can receive instantly. 

> Once funded, if you have not prompted users to backup their wallet prior to funding it’s important you prompt them to do so once it is. This is explained more in depth in the backing up a wallet section.

Reviewer 2: ```suggestion
Once funded, if you have not prompted the user to backup their wallet yet, it’s important you do so at this stage. Backups are explained more in-depth in the [backing up a wallet section]({{'/guide/onboarding/backing-up-a-wallet/' | relative_url}}).
```
